prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -182,37 +186,31 @@ def get_function_bullet_point_list(agent: Agent, model: str) -> str:
 
     st.divider()
     st.subheader("Configure:")
-    if not agent_is_initialized():
-        model = st.selectbox(
-            "Model",
-            [
-                "gpt-4-turbo",
-                "gpt-3.5-turbo-0125",
-                "gpt-4o-2024-05-13",
-            ],
-            index=0,
-        )
-        if model is None:
-            st.error("Please select a model.")
-        st.session_state.system_prompt_select = st.selectbox(
-            "Initial memory",
-            [p.value for p in SystemPromptChoice],
-            index=0,
-        )
-    else:
-        model = st.selectbox(
-            "Model",
-            [st.session_state.agent.llm.generator.model],
-            index=0,
-            disabled=True,
-        )
-        st.selectbox(
-            "Initial memory",
-            [st.session_state.system_prompt_select],
-            index=0,
-            disabled=True,
-        )
-    model = check_not_none(model)
+    st.selectbox(
```

Comment:
These look like nice changes (though not related to saving system prompt, right?). Another issue the app has on main is that the disabling of these select boxes also has a lag - the disabling happens when you run the agent for the second time, instead of at the end of the first run. Do you know if it's possible to move the disabling to after the agent has been run, to fix this?

Reply:
Great question - I don't know the answer, but when I do this (using "key" and reading from `session_state`), it works like a charm:
```
st.toggle("on_off", key="on_off")
st.write(f"on-off {st.session_state.on_off}")
st.button("will be deactivated", disabled=st.session_state.on_off)
```
So my guess is that the check `agent in st.session_state` is not updating properly, whereas the update is more "synchronous" when setting keys via streamlit widgets.